src/parser/mod.rs

```diff
@@ -5174,6 +5174,15 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let with = if self.parse_keyword(Keyword::WITH) {
```

Comment:
I'm wondering would it make more sense to have this limited to Postgres and Generic dialects? I'm guessing this functionality isn't common (took a quick look and found mysql has but the syntax is different so the logic won't be reusable if we later add support)

Reply:
Looks like mysql doesn't have `WITH` clause, it's `WITH PARSER`. https://dev.mysql.com/doc/refman/8.4/en/create-index.html
But I'm ok if this needs to be limited to Postgres and Generic dialects.

Reply:
Yeah I think we can limit it to the pg and generic dialects - mysql support will probably be the same in that we'd land in this code block since we need to first consume a WITH in both cases

Reply:
Done.